starlette/routing.py

```diff
@@ -773,7 +783,7 @@ def host(
     def add_route(
         self,
         path: str,
-        endpoint: typing.Callable,
+        endpoint: typing.Callable[..., typing.Any],
```

Comment:
Isn't an endpoint always `Callable[[Request], typing.Any]`?

Reply:
I went for `Callable[..., Any]` as this `endpoint` argument is passed to `routing.Route`, which is doing the following:

https://github.com/encode/starlette/blob/2168e47052239da5df35d5353bb986f760c51cef/starlette/routing.py#L209-L234

So it seems `endpoint` can also be a `ASGIApp` (+ there's a bunch of things related to `functools.partial`)